eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java

```diff
@@ -1038,17 +1038,40 @@ public V removeKey(K key)
     @Override
     public boolean removeIf(Predicate2<? super K, ? super V> predicate)
     {
-        int previousSize = this.size();
-        Iterator<Entry<K, V>> iterator = this.entrySet().iterator();
-        while (iterator.hasNext())
+        int previousOccupied = this.occupied;
+        for (int index = 0; index < this.table.length; index += 2)
         {
-            Entry<K, V> entry = iterator.next();
-            if (predicate.accept(entry.getKey(), entry.getValue()))
+            Object cur = this.table[index];
+            if (cur == null)
+            {
+                continue;
+            }
+            if (cur == CHAINED_KEY)
             {
-                iterator.remove();
+                Object[] chain = (Object[]) this.table[index + 1];
+                for (int chIndex = 0; chIndex < chain.length; chIndex += 2)
+                {
+                    K key = this.nonSentinel(chain[chIndex]);
```

Comment:
also, if `chain[chIndex] == null`, break the loop immediately.

Reply:
Thanks for Pointing this out. Fixed it.